### PR TITLE
feat: rewrite e2e test(ssl_test) with ginkgo

### PR DESCRIPTION
**Pull request by Jaycean**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
#1500

___
### New feature or improvement
-  Rewrite e2e test(ssl_test) with ginkgo

### CODE REVIEW

File: api/test/e2enew/ssl/ssl_test.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+				addr = "127.0.0.1:9443"
+			}
+			dialer := &net.Dialer{}
+			return dialer.DialContext(ctx, network, addr)
+		}
+		_, err = http.Get("https://www.test2.com:9443")
+		assert.NotNil(t, err)
+		assert.EqualError(t, err, "Get https://www.test2.com:9443: remote error: tls: internal error")
+		//create ssl fail - key and cert not match
+		base.RunTestCase(base.HttpTestCase{
```
Comment (nic-chen) [inline api/test/e2enew/ssl/ssl_test.go:74]: we could use table entry for these test cases.

Comment (Jaycean) [reply api/test/e2enew/ssl/ssl_test.go:74]: This use case requires the use of serialized SSL body incalidbody data.
Body data cannot be processed with table entry.
Do you have any good suggestions？

Comment (nic-chen) [reply api/test/e2enew/ssl/ssl_test.go:74]: oh, I see. ignore that.

Comment (Jaycean) [reply api/test/e2enew/ssl/ssl_test.go:74]: ok.

**Review by starsz — APPROVED** [review]

**Review by nic-chen — APPROVED** [review]

**Comment by nic-chen** [comment]
merged. thanks a lot! @Jaycean

**Pull request merged**